coordinator/src/db/polls.rs

```diff
@@ -0,0 +1,134 @@
+#[diesel(table_name = polls)]
+#[diesel(primary_key(id))]
+pub struct Poll {
+    pub id: i32,
+    pub poll_type: PollType,
+    pub question: String,
+    pub active: bool,
+    pub creation_timestamp: OffsetDateTime,
+}
+
+#[derive(Insertable, Queryable, Identifiable, Selectable, Debug, Clone, Eq, PartialEq)]
+#[diesel(belongs_to(Poll))]
+#[diesel(table_name = choices)]
+#[diesel(primary_key(id))]
+pub struct Choice {
+    pub id: i32,
+    pub poll_id: i32,
+    pub value: String,
+}
+#[derive(Insertable, Queryable, Identifiable, Debug, Clone)]
+#[diesel(primary_key(id))]
+pub struct Answer {
+    pub id: Option<i32>,
+    pub choice_id: i32,
+    pub trader_pubkey: String,
+    pub value: String,
+    pub creation_timestamp: OffsetDateTime,
+}
+
+pub fn active(conn: &mut PgConnection) -> QueryResult<Vec<commons::Poll>> {
+    let _polls: Vec<Poll> = polls::table.load(conn)?;
+    let _choices: Vec<Choice> = choices::table.load(conn)?;
+
+    let results = polls::table
+        .left_join(choices::table) //.on(polls::columns::id.eq(choices::columns::poll_id)))
+        .select(<(Poll, Option<Choice>)>::as_select())
+        .load::<(Poll, Option<Choice>)>(conn)?;
+
+    let mut polls_with_choices = HashMap::new();
+    for (poll, choice) in results {
+        let entry = polls_with_choices.entry(poll).or_insert_with(Vec::new);
+        if let Some(choice) = choice {
+            entry.push(choice);
+        }
+    }
+
+    let polls = polls_with_choices
+        .into_iter()
+        .map(|(poll, choice_vec)| commons::Poll {
+            id: poll.id,
+            poll_type: poll.poll_type.into(), // TODO: fix me
+            question: poll.question,
+            choices: choice_vec
+                .into_iter()
+                .map(|choice| commons::Choice {
+                    id: choice.id,
+                    value: choice.value,
+                })
+                .collect(),
+        })
+        .collect();
+    Ok(polls)
+}
+
+impl From<PollType> for commons::PollType {
+    fn from(value: PollType) -> Self {
+        match value {
+            PollType::SingleChoice => commons::PollType::SingleChoice,
+        }
+    }
+}
+
+pub fn add_answer(conn: &mut PgConnection, answers: commons::PollAnswers) -> Result<()> {
+    let mut affected_rows = 0;
+    for answer in answers.answers {
+        affected_rows += diesel::insert_into(answers::table)
+            .values(Answer {
+                id: None,
+                choice_id: answer.choice_id,
+                trader_pubkey: answers.trader_pk.to_string(),
+                value: answer.value,
+                creation_timestamp: OffsetDateTime::now_utc(),
+            })
+            .execute(conn)?;
+    }
+
+    if affected_rows == 0 {
+        bail!(
+            "Could not insert answers by user {}.",
+            answers.trader_pk.to_string()
+        );
+    } else {
+        tracing::debug!(%affected_rows, "Added new answers to a poll.");
```

Comment:
🔧 I guess `trace` is better suited for that log. I would also add the `trader_pk`.